FreeRTOS/FreeRTOS-Kernel
# Update system call entry mechanism

Description
-----------
Earlier the System Call entry from an unprivileged task looked like:
1. SVC for entering system call.
2. System call implementation.
3. SVC for exiting system call.

Now, the system call entry needs to make only one system call and everything else is handled internally.

This PR also makes the following small changes:
1. Add one struct param for system calls with 5 parameters. This removes the need for special handling for system calls with 5 parameters.
2. Remove raise privilege SVC when MPU wrapper v2 is used.
3. Add additional run time parameter checks to MPU wrappers for xTaskGenericNotify and xQueueTakeMutexRecursive APIs.

Test Steps
-----------
Tested on the following platforms:
1. STM32H743ZI (Cortex-M7)
2. STM32L152RE (Cortex-M3)
3. Nuvoton M2351 (Cortex-M23)
4. NXP LPC55S69 (Cortex-M33)

Checklist:
----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
